eval.go

```diff
@@ -602,7 +603,11 @@ func (st *Runtime) evalPrimaryExpressionGroup(node Expression) reflect.Value {
 					node.errorf("%s is not assignable|convertible to map key %s", indexType.String(), key.String())
 				}
 			}
-			return baseExpression.MapIndex(indexExpression)
+			baseExpression = baseExpression.MapIndex(indexExpression)
+			if baseExpression.Kind() == reflect.Interface && baseExpression.IsNil() {
+				return emptyString
```

Comment:
I don't think it's a good idea to return a string here regardless of the underlying map's value type.

IMO, when a map contains an actual `nil` value for a key, then that value should be returned as-is to the caller.

We use map access syntax a lot to work on JSON objects (deserialized into map[string]interface{}), and checking `m["someKey"] == nil` is a lot clearer when expecting a `null` in the original JSON data than `m["someKey"] == ""` or `isset(m["someKey"]` (which would also be false if the key doesn't exist in the map at all).

When a map doesn't contain the indexExpression as key, `baseExpression.MapIndex(indexExpression)` will return `reflect.Value{}` and either that should be returned, to signal to the runtime that the key was not found, or the zero value of the map's value type should be returned (`reflect.Zero(baseExpression.Type().Elem())`), but then the runtime can never be sure if the value came from the map or if the key didn't exist.

Reply:
Returning the zero value of the map's value type means we can call methods etc. without any error when the map is a `map[string]User` (i.e. value type is a custom struct type), which could hide/shadow missing data.

Reply:
Got it, sounds very reasonable when you explain it like that. 😅